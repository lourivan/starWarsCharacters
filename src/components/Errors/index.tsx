import React from 'react'
import { View } from 'react-native'

// import { Container } from './styles';

export const Errors: React.FC = () => {
  return <View />
}
