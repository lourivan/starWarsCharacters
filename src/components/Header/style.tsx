import styled from 'styled-components/native'

export const Container = styled.View`
  background-color: ${({ theme }) => theme.colors.primary};
  align-items: center;
  padding-top: 50px;
  padding-bottom: 20px;
`
export const ImageHeader = styled.Image.attrs({
  resizeMode: 'contain',
})`
  width: 156px;
  height: 88px;
`
