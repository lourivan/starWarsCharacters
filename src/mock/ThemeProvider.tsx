import React from 'react'
import { ThemeProvider } from 'styled-components/native'
import { theme } from '../'

interface IChildrenProps {
  children: any
}

export const ThemeProviderMock: React.FC<IChildrenProps> = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}
