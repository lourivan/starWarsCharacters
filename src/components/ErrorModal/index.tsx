import React from 'react'
import { Modal } from 'react-native'
import LottieView from 'lottie-react-native'

import * as S from './style'

export const ErrorModal = (props) => {
  const { isError, onRetry } = props
  return (
    <Modal transparent={true} animationType={'none'} visible={isError}>
      <S.Container>
        <LottieView
          source={require('assets/lottie/error-animation.json')}
          autoPlay
          loop={false}
          style={{ width: 200, height: 200 }}
        />
        <S.ErrorDescription>
          <S.ErrorText>Oops, Something Went Wrong</S.ErrorText>
          <S.ButtonSkip color={'#000'} onPress={onRetry} mode='contained'>
            Retry
          </S.ButtonSkip>
        </S.ErrorDescription>
      </S.Container>
    </Modal>
  )
}
