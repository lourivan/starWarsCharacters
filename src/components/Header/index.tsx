import React, { useMemo } from 'react'
import { Image } from 'react-native'
import { Appbar } from 'react-native-paper'

import * as S from './style'
import Logo from 'assets/images/Logo.png'
export interface HeaderProps {
  type?: 'default' | 'page'
  onBack?: () => void
  title?: string
  subtitle?: string
}

export const Header: React.FC<HeaderProps> = ({
  type,
  onBack,
  title,
  subtitle,
}) => {
  const TAG = 'Header_component'
  const DefaultHeader = () => (
    <S.Container>
      <S.ImageHeader
        testID={`${TAG}_default_image`}
        source={Image.resolveAssetSource(Logo).uri}
      />
    </S.Container>
  )

  const PageHeader = () => (
    <Appbar.Header>
      <Appbar.BackAction testID={`${TAG}_button_back`} onPress={onBack} />
      <Appbar.Content
        title={title}
        subtitle={subtitle}
        titleStyle={{ fontSize: 20 }}
        subtitleStyle={{ fontSize: 16 }}
      />
    </Appbar.Header>
  )

  const render = useMemo(
    () => (type != 'default' ? <PageHeader /> : <DefaultHeader />),
    [type, title, subtitle, onBack],
  )

  return render
}
