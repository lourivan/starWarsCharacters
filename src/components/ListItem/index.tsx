import React from 'react'

import * as S from './style'
import { assignTestId } from '~/utils/qualityAssurance'

export interface ItemProps {
  title: string
  description?: string
  onPress?: () => void
}

export const ListItem: React.FC<ItemProps> = ({
  title,
  description = '',
  onPress,
}) => {
  const TAG = 'listItem'
  return (
    <S.Item
      {...assignTestId('TouchableOpacity', TAG)}
      title={title}
      description={description}
      right={(_props) => <S.IconRight />}
      onPress={onPress}
    />
  )
}
