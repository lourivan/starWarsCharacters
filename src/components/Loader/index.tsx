import React from 'react'
import { Modal } from 'react-native'
import LottieView from 'lottie-react-native'

import * as S from './style'

export const Loader = (props) => {
  const { loading } = props
  return (
    <Modal transparent={true} animationType={'none'} visible={loading}>
      <S.ModalBackground>
        <LottieView
          source={require('assets/lottie/star-wars.json')}
          autoPlay
          loop
          style={{ width: 200, height: 200 }}
        />
      </S.ModalBackground>
    </Modal>
  )
}
