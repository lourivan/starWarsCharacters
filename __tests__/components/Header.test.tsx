import React from 'react'
import { render, fireEvent } from '@testing-library/react-native'

import { Header, HeaderProps } from '../../src/components/Header'
import { ThemeProviderMock } from '../../src/mock/ThemeProvider'

describe('Components/Header', () => {
  let props: HeaderProps = {
    type: 'default',
    onBack: jest.fn(),
  }

  let pageProps: HeaderProps = {
    type: 'page',
    title: 'Page',
    subtitle: 'subtitle',
  }

  const defaultHeaderId = 'Header_component_default_image'

  it('renders Header with default props', async () => {
    const { getByTestId } = render(
      <ThemeProviderMock children={<Header {...props} />} />,
    )
    expect(getByTestId('Header_component_default_image')).not.toBeNull()
  })

  it('renders Header with type props', async () => {
    const page = { ...props, ...pageProps }
    const { queryByText } = render(
      <ThemeProviderMock children={<Header {...page} />} />,
    )
    expect(queryByText('Page')).not.toBeNull()
    expect(queryByText('subtitle')).not.toBeNull()
  })

  it('fire event onBack', async () => {
    const page = { ...props, ...pageProps }
    const { getByTestId } = render(
      <ThemeProviderMock children={<Header {...page} />} />,
    )
    fireEvent.press(getByTestId('Header_component_button_back'))
    expect(props.onBack).toHaveBeenCalledTimes(1)
  })

  it('should render snapshot', () => {
    const { toJSON } = render(
      <ThemeProviderMock children={<Header {...props} />} />,
    )
    expect(toJSON()).toMatchSnapshot()
  })
})
