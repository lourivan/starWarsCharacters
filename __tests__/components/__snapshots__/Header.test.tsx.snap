// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Header should render snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#000",
        "paddingBottom": 20,
        "paddingTop": 50,
      },
    ]
  }
>
  <Image
    resizeMode="contain"
    style={
      Array [
        Object {
          "height": 88,
          "width": 156,
        },
      ]
    }
    testID="Header_component_default_image"
  />
</View>
`;
